lab-miketreat/lib/modules.js

```diff
@@ -0,0 +1,45 @@
+'use strict';
+
+module.exports = exports = {};
+
+
+exports.SpaceStationModule = function(name, role){
+  this.name = name;
+  this.role = role;
+};
+
+// call example
+exports.LifeSupport = function(name, role){
+  exports.SpaceStationModule.call(this, name, role);
+  this.vitalSystems = ['Air', 'Water'];
+};
+
+// apply example
+exports.Navigation = function(name, role){
+  exports.SpaceStationModule.apply(this, [name, role]);
+  this.vitalSystems = ['Auto-Pilot', 'NAV-COM'];
+};
+
+
+// bind example
+
+// Sets up initial function currying using this keywords.
+exports.alertContents = function() {
+  return `There are ${this.snakemonsters} Snake Monster(s), ${this.ewoks} Ewok(s), and ${this.humans} Human(s) currently in the Garbage-compactor!!`;
+};
+
+exports.Waste = function(name, role){
+  exports.SpaceStationModule.call(this, name, role );
+  this.vitalSystems = ['Garbage-compactor', 'Poop-dehydrationatior' ];
+
+  // this sets up the weird ass function passing from the initial function. I also added a setBoundFunc function so that you could dynamically pass in context as an arg.
+  var testFunc = exports.alertContents;
+  let setBoundFunc = function(ctx){
+    let boundFunc = testFunc.bind(ctx);
```

Comment:
Great example of bind. I'm glad you noticed the relation of currying and .bind()